Fix NU block description

johannmartinet opened the pull request:
Fix NU block description for following cases :
- Either prompt or total nu is given
- Both prompt and total nu are given

Recompile and regenerate PDF.

Related issue : #12

johannmartinet commented:
Leads to following changes in the PDF : 

![1572533632](https://user-images.githubusercontent.com/47329447/67958244-29e6db80-fbf7-11e9-92a0-f56ee9f3ac98.png)

jlconlin commented:
So I think this is a little confusing. I'd rather see another level of nesting such as:

There are two posibilities for the NU Block:
1. No NU Block (JXS(2) = 0)
2. NU Block (JXS(2) != 0)
    The NU array begins at location ...
    - Prompt *or* total nubar is given (XSS(JXS(2)) > 0
       ...
    - Both prompt *and* total nubar are given (XSS(JXS(2))< 0
       ...
  
I think this way would make it a little more clear. 

I could make the changes, but I'm not sure how to do that within this Pull Request.

jlconlin commented on src/ContinuousEnergyNeutron.tex:
```diff
@@ -113,9 +113,9 @@ \subsubsection{\textsf{NU} Block}\label{sec:NUBlock}
 \begin{enumerate}
   \item {\bfseries\sffamily No NU Block.} (\jxs{2}=0)
-  \item {\bfseries\sffamily Either prompt or total \nubar\ is given (but not both).} (\jxs{2} > 0) \\
+  \item {\bfseries\sffamily Either prompt or total \nubar\ is given (but not both).} (\jxs{2}$\neq0$ and \xss{\jxs{2}} > 0) \\
```
Please add an additional level of nesting as suggested

paulromano commented:
Rather than saying JXS(2) != 0, shouldn't we say JXS(2) > 0? Doesn't look like it should ever be negative if it's used as an index in XSS

jlconlin commented:
Yeah, good point @paulromano It shouldn't ever be negative. A negative value doesn't even make sense.

johannmartinet commented:
@jlconlin @paulromano Thanks for your comments. I concur with you, I will add an additional level of nesting and I think `JXS(2) > 0` is more clear.

johannmartinet commented:
New NU block description 

<img width="615" alt="Screenshot 2019-11-01 at 23 02 47" src="https://user-images.githubusercontent.com/47329447/68059204-da3e0800-fcfb-11e9-8d4e-f0f16cd66083.png">

johannmartinet commented:
With the new changes, I have a new comment : 
Is it useful to introduce with 

> There are two possibilities for the NU Block: 
> 1. No NU Block. (JXS(2)=0) 

and the case where there is no NU block is given in the introduction of section 4 

> 2. NU Block—contains prompt, delayed and/or total ν as a function of incident neutron energy. The NU Block exists only for fissionable isotopes; that is, if JXS(2)̸= 0. See Section 4.3.2.

For example the description of the MTR block does not include the case where MTR block does not exist (NXS(4) = 0) ...

jlconlin commented:
@johannmartinet I don't quite understand what you are getting at with your latest comment. Can you clarify a bit more?

whaeck commented on src/ContinuousEnergyNeutron.tex:
```diff
@@ -110,16 +110,19 @@ \subsubsection{\textsf{ESZ} Block}\label{sec:ESZBlock}
+  \item {\bfseries\sffamily NU Block exists.} (\jxs{2} > 0)
+  \begin{itemize}
+	\item {\bfseries\sffamily Prompt or total \nubar\ is given (but not both).} (\xss{\jxs{2}} > 0) \\
+	The \aceArray{NU} array begins at location \xss{KNU} where \var{KNU}=\jxs{2} + 1.
```
From what I can see in the NJOY source code (see subroutine change in acefc.f90), when either prompt or total is given, LNU is stored in ```\xss{\jxs{2}}``` so that in this case ```\var{KNU}=\jxs{2}```

johannmartinet replied:
I concur with this!

whaeck commented on src/ContinuousEnergyNeutron.tex:
```diff
@@ -110,16 +110,19 @@ \subsubsection{\textsf{ESZ} Block}\label{sec:ESZBlock}
+  \begin{itemize}
+	\item {\bfseries\sffamily Prompt or total \nubar\ is given (but not both).} (\xss{\jxs{2}} > 0) \\
+	The \aceArray{NU} array begins at location \xss{KNU} where \var{KNU}=\jxs{2} + 1.
+	\item {\bfseries\sffamily Both prompt and total \nubar\ are given.} (\xss{\jxs{2}} < 0)
```
In this case, ```\xss{\jxs{2}}``` is a locator for the total nu array. This value is not present when either prompt or total is given. The quoted values for ```\var{KNU}``` given below are therefore correct.

We should maybe rewrite this piece of the ACE specification to better reflect the presence of the locator.

johannmartinet replied:
This is a good point, this locator should be emphasized.

whaeck commented:
As an alternative, I have created a branch fix/nu

I have rewritten the NU block (taking into account the changes proposed in this pull request), added the DNU block and started work on the BDD block (precursor information). I still need to add the DNEDL and DNED blocks.

johannmartinet commented:
> @johannmartinet I don't quite understand what you are getting at with your latest comment. Can you clarify a bit more?

I mean (in section 4.3.2)  the case `JXS(2) = 0` is redundant with the introduction of section 4 :

> NU Block—contains prompt, delayed and/or total ν as a function of incident neutron energy. The NU Block exists only for fissionable isotopes; that is, if JXS(2) != 0. See Section 4.3.2.

A counterexemple is the MTR block (section 4.3.3). The block description does not introduce this redundancy (no MTR block when NXS(4) = 0).

Is it necessary to introduce NU block description (section 4.3.2) with the case where it does not exist ? If yes, should we do the same with other block descriptions ? 
Hope it clarify a bit more ...

johannmartinet commented:
> As an alternative, I have created a branch fix/nu
> 
> I have rewritten the NU block (taking into account the changes proposed in this pull request), added the DNU block and started work on the BDD block (precursor information). I still need to add the DNEDL and DNED blocks.

@whaeck I will check PR #14

whaeck commented:
Closed because of #14
